voice-search/voice-search-impl/src/main/java/com/duckduckgo/voice/impl/listeningmode/VoiceSearchActivity.kt

```diff
@@ -0,0 +1,161 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.voice.impl.listeningmode
+
+import android.annotation.SuppressLint
+import android.app.Activity
+import android.content.Intent
+import android.graphics.Color
+import android.graphics.drawable.ColorDrawable
+import android.os.Build
+import android.os.Bundle
+import androidx.core.view.postDelayed
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.flowWithLifecycle
+import androidx.lifecycle.lifecycleScope
+import com.duckduckgo.anvil.annotations.InjectWith
+import com.duckduckgo.app.global.DuckDuckGoActivity
+import com.duckduckgo.voice.impl.listeningmode.VoiceSearchViewModel.Command
+import com.duckduckgo.voice.impl.listeningmode.ui.VoiceRecognizingIndicator.Action.INDICATOR_CLICKED
+import com.duckduckgo.voice.impl.listeningmode.ui.VoiceRecognizingIndicator.Model
+import com.duckduckgo.appbuildconfig.api.AppBuildConfig
+import com.duckduckgo.di.scopes.ActivityScope
+import com.duckduckgo.mobile.android.ui.viewbinding.viewBinding
+import com.duckduckgo.voice.impl.R
+import com.duckduckgo.voice.impl.databinding.ActivityVoiceSearchBinding
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import java.util.Locale
+import javax.inject.Inject
+
+@InjectWith(ActivityScope::class)
+class VoiceSearchActivity : DuckDuckGoActivity() {
+    companion object {
+        const val EXTRA_VOICE_RESULT = "extra.voice.result"
+        const val DELAY_SPEAKNOW_REMINDER_MILLIS = 2000L
+    }
+
+    @Inject
+    lateinit var appBuildConfig: AppBuildConfig
+
+    private val viewModel: VoiceSearchViewModel by bindViewModel()
+    private val binding: ActivityVoiceSearchBinding by viewBinding()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        overridePendingTransition(0, 0)
+        makeBackgroundTransparent()
+        setContentView(binding.root)
+        configureToolbar()
+        configureViews()
+    }
+
+    private fun configureViews() {
+        binding.indicator.onAction {
+            if (it == INDICATOR_CLICKED) {
+                viewModel.userInitiatesSearchComplete()
+            }
+        }
+        binding.speechResults.let {
+            it.postDelayed(DELAY_SPEAKNOW_REMINDER_MILLIS) {
+                if (it.text.isEmpty()) it.text = getString(R.string.voiceSearchListening)
+            }
+        }
+    }
+
+    private fun configureToolbar() {
+        setupToolbar(binding.toolbar)
+        supportActionBar?.title = ""
+    }
+
+    @SuppressLint("NewApi")
+    private fun makeBackgroundTransparent() {
+        if (appBuildConfig.sdkInt >= Build.VERSION_CODES.S) {
+            window.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
+            setTranslucent(true)
+        }
+    }
+
+    override fun onStart() {
+        super.onStart()
+        observeViewModel()
```

Comment:
In other parts of the code we observe the ViewModel in `OnCreate()`, is there a reason for using `onStart()` in this case?

Reply:
Ah I think I incorrectly placed it here. I don't think it was intentional that I put it in onStart()